pkg/kubectl/cmd/annotation.go

```diff
@@ -0,0 +1,278 @@
+/*
+Copyright 2014 The Kubernetes Authors All rights reserved.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package cmd
+
+import (
+	"bytes"
+	"fmt"
+	"io"
+	"strings"
+
+	"github.com/GoogleCloudPlatform/kubernetes/pkg/api"
+	cmdutil "github.com/GoogleCloudPlatform/kubernetes/pkg/kubectl/cmd/util"
+	"github.com/GoogleCloudPlatform/kubernetes/pkg/kubectl/resource"
+	"github.com/GoogleCloudPlatform/kubernetes/pkg/runtime"
+	"github.com/spf13/cobra"
+)
+
+// AnnotateOptions have the data required to perform the annotate operation
+type AnnotateOptions struct {
+	out               io.Writer
+	resources         []string
+	newAnnotations    map[string]string
+	removeAnnotations []string
+	builder           *resource.Builder
+
+	overwrite       bool
+	all             bool
+	resourceVersion string
+}
+
+const (
+	annotate_long = `Update the annotations on one or more resources.
+
+An annotation is a key/value pair that can hold larger (compared to a label), and possibly not human-readable, data.
+It is intended to store non-identifying auxiliary data, especially data manipulated by tools and system extensions.
+If --overwrite is true, then existing annotations can be overwritten, otherwise attempting to overwrite an annotation will result in an error.
+If --resource-version is specified, then updates will use this resource version, otherwise the existing resource-version will be used.
+
+Possible resources include (case insensitive): pods (po), services (svc),
+replicationcontrollers (rc), nodes (no), events (ev), componentstatuses (cs),
+limitranges (limits), persistentvolumes (pv), persistentvolumeclaims (pvc),
+resourcequotas (quota) or secrets.`
+	annotate_example = `# Update pod 'foo' with the annotation 'description' and the value 'my frontend'.
+$ kubectl annotate pods foo description='my frontend'
+
+# Update pod 'foo' with the annotation 'description' and the value 'my frontend running nginx', overwriting any existing value.
+$ kubectl annotate --overwrite pods foo description='my frontend running nginx'
+
+# Update all pods in the namespace
+$ kubectl annotate pods --all description='my frontend running nginx'
+
+# Update pod 'foo' only if the resource is unchanged from version 1.
+$ kubectl annotate pods foo description='my frontend running nginx' --resource-version=1
+
+# Update pod 'foo' by removing an annotation named 'source_id' if it exists.
+# Does not require the --overwrite flag.
+$ kubectl annotate pods foo description-`
+)
```

Comment:
It should be 'description'. I changed the example and forgot to change the comment.